dev/io.openliberty.data.internal.persistence/src/io/openliberty/data/internal/persistence/cdi/RepositoryProducer.java

```diff
@@ -103,6 +121,39 @@ public void dispose(R repository) {
         handler.beanDisposed();
     }
 
+    /**
+     * Error handling for a timeout that occurs when attempting to obtain an
+     * EntityManagerBuilder.
+     *
+     * @param repositoryInterface the repository interface.
+     * @param exception           the TimeoutException.
+     * @return DataException with an appropriate error message.
+     */
+    @Trivial
+    private DataException excTimedOut(Class<?> repositoryInterface, Throwable exception) {
+        if (CheckpointPhase.getPhase() == CheckpointPhase.INACTIVE) {
```

Comment:
The CheckpointPhase is going to always be the same enum for the duration of the server instance.  That is it will always be either `AFTER_APP_START`, `BEFORE_APP_START`, or `INACTIVE`.  This is set right at the launch of Liberty to the value and it never will change, even after restore.  This should instead check the `CheckpointPhase.restored()` method
```suggestion
        if (CheckpointPhase.getPhase().restored()) {
```

This will always return `true` for "normal" launches when the value is `INACTIVE`.  It will return `false` for the other values if we are currently before a checkpoint has happened.  There has been discussion in the past about adding a method `beforeCheckpoint` that would only return `true` if before the checkpoint has happened.  That may make more sense when thinking about the `INACTIVE` value because such a value would never have a before checkpoint state.  But for now we only have the `restored` method.